artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/AddressSettings.java

```diff
@@ -505,6 +505,9 @@ public long getPageSizeBytes() {
    }
 
    public AddressSettings setPageSizeBytes(final long pageSize) {
+      if (pageSize > Integer.MAX_VALUE) {
+         throw new IllegalArgumentException("pageSize must be < " + Integer.MAX_VALUE);
```

Comment:
Why not change long to int, this would then be cleaner from anyone trying to use the method

Reply:
The AddressSettings maybe persisted in disk, and if we change long to int, we would incorrectly decode it. Does it make sense?

Reply:
Not really. If you now decode something thats larger than int it will blow up with the exception youve added. No different. Except we are actually making the method take the correct primitive number

Reply:
Also should update the schema to int so then when people do parse configs via xsd validators it would fail